alsa-ucm-conf: 1.2.15.1 → 1.2.15.3-unstable-2026-01-29; validate in alsa-lib

illdefined opened the pull request:
Switch `alsa-ucm-conf` to rolling release.

Updates to the UCM configuration files are released in tandem with the ALSA library every half year or so.

The UCM configuration files however receive frequent updates with new profiles for additional hardware and I believe that users would benefit from making these available in nixpkgs independently of the ALSA lib release cycle.

The configuration files are validated automatically by upstream after review and so I don’t expect this to break very often.
Just to make sure, I added validation of the UCM configuration to `alsa-lib`, as part of `installCheckPhase`.



<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

illdefined committed:
alsa-ucm-conf: 1.2.15.1 -> 1.2.15.3-unstable-2026-01-29

illdefined committed:
alsa-lib: validate UCM configuration

roastiek reviewed (approved):
Yeah, it is probably better. Other distributions are doing something similar, too.

One optional thing. Maybe add some check for any new `exec` option in new releases, so they will not be missed when updating. 

I have come with `if grep -E 'exec\s+"[-]?/s?bin/' -r $out; then echo "new unattended exec found"; exit 1; fi`.

illdefined committed:
alsa-ucm-conf: check for unattended exec directives

illdefined committed:
alsa-ucm-conf: replace executable paths more flexibly

Instead of replacing executable paths in specific files, search for exec
directives in all configuration files, replacing executables with their
corresponding absolute store paths according to an attribute set,
falling back to ‘false’ for packages unavailable on the host platform.

roastiek reviewed (approved):
The `sed` expressions in the final `postPatch` are still easy to understand.

And when I noticed the usage of `finalAttrs.executablePkgs`, I made sense of why you had written the nix expression as it is.

illdefined committed:
alsa-ucm-conf: Use nhlt-dmic-info from alsa-utils-nhlt

Now that the nhlt-dmic-info tool was split off from alsa-utils into a
separate package, we can safely refer to it without producing a
circular dependency.
